src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -695,6 +696,10 @@ ssize_t sockinfo_tcp::tx(const tx_call_t call_type, const struct iovec* p_iov, c
 	si_tcp_logfunc("tx: iov=%p niovs=%d dummy=%d", p_iov, sz_iov, is_dummy);
 	lock_tcp_con();
 
+	if (unlikely(MCE_SPEC_STAC == m_mce_spec)) {
```

Comment:
1. You are right, it shouldn't be locked.
2. I will try to find an alternative solution.